### DISCUSSION TITLE
Standard way of getting list of classes for a model - exporting to ONNX

### DISCUSSION THREAD

**Discussion by wereii** [ultralytics/yolov5#11240] [Q&A] [answered]
Hey,
as far as I could find, onnx models do not contain labels/classes.

What would be the best/standard way to get the labels/class names from a yolo .pt model?
Sorry if I am mixing something up.

**Comment by wereii** [comment, marked as answer]
I guess I've just found the labels `model.names`.

```py
# requires onnx>=1.12.0
from ultralytics import YOLO
import json

model = YOLO("yolov8n.pt")
model.export(format="onnx")


with open("yolov8n.labels.json", "w") as labels_file:
    data = {"labels": model.names}
    json.dump(data, labels_file)
```

**Comment by cloudlakecho** [comment]
@wereii Do you know the method for "ONNX" file type? 

I tried this snippet from [Stack Overflow](https://stackoverflow.com/a/76864106/5595995):
```
import onnx,ast
m = onnx.load(<ONNX file>)
props = { p.key : p.value for p in m.metadata_props }
if 'names' in props:
    names = ast.literal_eval(props['names'])
```
but my array is empty.
